src/dev/impl/DevToys/Views/Tools/Text/MarkdownPreview/MarkdownPreviewToolPage.xaml.cs

```diff
@@ -65,5 +69,25 @@ public void Receive(NavigateToMarkdownPreviewHtmlMessage message)
             Arguments.NotNull(message, nameof(message));
             PreviewWebView.NavigateToString(message.Html);
         }
+
+        private async void CopyButton_Click(object sender, RoutedEventArgs e)
```

Comment:
* I think by MVVM convention this is probably an anti-pattern? That is, to have the click handler in the view?
* Also it feels a little silly to have to extract the HTML body from the webview rather than having access to the converted HTML. An alternative I considered was to make that property accessible to the view instead of doing what I did here. However, I wasn't sure of the convention of accessing the ViewModel's property from the View?

Reply:
Hello,

Thanks for your contribution. 
Yes, this is an anti-pattern. The approach with MVVM is to make a command. In short, the button would be directly linked to something in the view model.
As an example, you can look at UUID Generator's "GenerateCommand", which is called when clicking the "Generate" button. https://github.com/veler/DevToys/blob/e2c9b71afd91c3f8135ea4bb8480c924773a3782/src/dev/impl/DevToys/ViewModels/Tools/Generators/GuidGenerator/GuidGeneratorToolViewModel.cs#L152-L152

You can look in the corresponding XAML how it is used.

For retrieving the HTML to copy, my advice is to generate the HTML itself, without trying to access the WebView (best practice is that the ViewModel doesn't interact directly with the controls in the UI).
For this, you can create a private method with this code:
https://github.com/veler/DevToys/blob/e2c9b71afd91c3f8135ea4bb8480c924773a3782/src/dev/impl/DevToys/ViewModels/Tools/Text/MarkdownPreview/MarkdownPreviewToolViewModel.cs#L137-L138
And pass the property `InputValue` in argument. Once you get the result of this method, you can copy it to the clipboard.
It will give you the exact same HTML :)

Reply:
@veler thanks for the info. Let me give that a shot. It'll be a good learning experience for me.